Documentation/Administration/Deployment/Automated/Index.rst

```diff
@@ -0,0 +1,195 @@
+:navigation-title: CI/CD Automation
+
+..  include:: /Includes.rst.txt
+..  _ci-cd-for-typo3-projects:
+
+==============================================
+CI/CD: Automatic deployment for TYPO3 Projects
+==============================================
+
+**Continuous Integration (CI)** and **Continuous Deployment/Delivery (CD)**
+are development practices that automate the process of building, testing,
+and deploying code. Implementing CI/CD for TYPO3 projects ensures higher
+quality releases, faster feedback loops, and lower risk of introducing bugs.
+
+..  contents::
+
+..  _why-ci-cd-for-typo3:
+
+Why CI/CD for TYPO3?
+====================
+
+TYPO3 is a powerful, enterprise-level CMS written in PHP. TYPO3 projects
+often involve custom extensions, configuration management (TypoScript, YAML config),
+and complex deployment workflows. Manual deployment increases the risk of
+human error, environment inconsistencies, and delayed releases. CI/CD
+automates these concerns.
+
+..  _ci-cd-common-stages:
+
+Common CI/CD Stages in TYPO3 Projects
+=====================================
+
+..  _ci-cd-code-quality-checks:
+
+Code Quality Checks
+-------------------
+
+*   **PHP Linting** (for example, :bash:`php -l`)
+*   `Code style tests and fixing <https://docs.typo3.org/permalink/t3coreapi:testing-projects-configuration-cs>`_
+*   **Static Analysis** (for example `PHPstan <https://docs.typo3.org/permalink/t3coreapi:testing-projects-configuration-phpstan>`_ or Psalm)
+*   Linting of other used formats like `TypoScript <https://github.com/martin-helmich/typo3-typoscript-lint>`_, XML, and YAML.
+
+..  _ci-cd-unit-functional-testing:
+
+Unit and Functional Testing
+---------------------------
+
+*   `Unit tests <https://docs.typo3.org/permalink/t3coreapi:testing-writing-unit>`_
+*   `Functional tests <https://docs.typo3.org/permalink/t3coreapi:testing-writing-functional>`_
+*   `Acceptance tests <https://docs.typo3.org/permalink/t3coreapi:testing-writing-acceptance>`_
+
+..  _ci-cd-building-artifacts:
+
+Building Artifacts
+------------------
+
+*   Installing required extensions and other PHP packages via Composer
+*   Compile frontend assets (e.g., SCSS, JavaScript) using tools like
+    Webpack, Gulp or Vite.
+
+..  _ci-cd-deployment:
+
+Deployment
+----------
+
+*   **File Synchronization**: Deploy code and assets using tools like `Rsync <https://docs.typo3.org/permalink/t3coreapi:deployment-rsync>`_,
+    `Deployer <https://docs.typo3.org/permalink/t3coreapi:deployment-deployer>`_,
```

Comment:
Deployer should be mentioned separately here since it is general deployment workflow tool, not limited to file sync.

Also see https://docs.typo3.org/m/typo3/reference-coreapi/main/en-us/Administration/Deployment/Tools/Deployer/Index.html

Reply:
Deployer has its own chapter. it is only mentioned briefly here as when you are using deployer you dont directly notice you use Rsync under the hood